File: backend/src/main/java/com/vira/dply/mapper/EnvironmentMapper.java
package com.vira.dply.mapper;

import org.mapstruct.Mapper;

import com.vira.dply.dto.EnvironmentDto;
import com.vira.dply.entity.EnvironmentEntity;

@Mapper(componentModel = "spring")
public interface EnvironmentMapper {

    EnvironmentDto toDto(EnvironmentEntity entity);

    EnvironmentEntity toEntity(EnvironmentDto dto);
}

File: backend/src/main/kotlin/com/vira/dply/mapper/EnvironmentMapper.kt
package com.vira.dply.mapper

import com.vira.dply.dto.EnvironmentDto
import com.vira.dply.dto.NewEnvironmentDto
import com.vira.dply.model.Environment
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface EnvironmentMapper {

    fun toEntity(dto: NewEnvironmentDto): Environment

    fun toDto(entity: Environment): EnvironmentDto
}